File: WebApp/ClientApp/src/Entities/Sales/Sale.ts
export interface Sale {
    id: string;
    customerId: string;
    salesPersonId: string;
    ProductId: string;
    salesDate: Date;
}

export type ServerSale = Omit<Sale, "salesDate"> & { salesDate: string }

export function deserializeSale(serverResponse:ServerSale):Sale {
    return {
        ...serverResponse,
        salesDate: new Date(serverResponse.salesDate)
    }
}

File: Domain/BeSpoked.BLL/Sales/Entities/Sale.cs
using BeSpoked.Common.EntityService.Entities;
using BeSpoked.Customers.Entities;
using BeSpoked.Products.Entities;
using BeSpoked.SalesTeam.Entities;

namespace BeSpoked.Sales.Entities;

public record Sale : Entity
{
    public required Guid CustomerId { get; set; }
    public required Guid SalesPersonId { get; set; }
    public required Guid ProductId { get; set; }
    public required DateTime SalesDate { get; set; }
    public required decimal CommissionAmount { get; set; }
    public required decimal SalesPrice { get; set; }
    public virtual SalesPerson SalesPerson { get; set; } = null!;
    public virtual Customer Customer { get; set; } = null!;
    public virtual Product Product { get; set; } = null!;
};